src/System.Private.ServiceModel/tools/test/Bridge/WcfTestBridgeCommon/PortManager.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+
+namespace WcfTestBridgeCommon
+{
+    // This class exists to create and delete firewall rules to
+    // manage which ports are open on behalf of the Bridge.
+    public class PortManager
+    {
```

Comment:
PortManager should have a mechanism (using finalization) to cause the ports to be closed on AppDomain shutdown.